### ISSUE TITLE
[FeatureRequest] Crome plugin\autotyping password

### ISSUE THREAD

**Issue by bademux** [keeweb/keeweb#4]
Hi,
Can you please distribute keeweb as chrome plugin with autotyping password into form?
Using clipboard is unsecure,
Thanks

**Comment by connor4312** [comment]
This could probably be done with [RobotJS](https://github.com/octalmage/robotjs) pretty easily... send a control+tab on Windows/Linux or command+tab on OSX, then type stuff in. I might give this a shot and PR this evening.

**Comment by antelle** [comment]
Chrome plugin should have different ui, more adopted to chrome. Using this ui directly in chrome will not work. In next releases, I'll add keepass-http to apps, to allow work with existing keepass browser extensions.

**Comment by vla7** [comment]
I subscribe to this FeatureRequest. Already created the same in passifox issues: https://github.com/pfn/passifox/issues/438

**Comment by antelle** [comment]
Keepass-http will be implemented here natively, so I think existing browser plugins will be easy to integrate. Anyway, I'm ready to discuss ways of integration, e.g. open some interface for plugin.

**Comment by paul-alkhimov** [comment]
Just my two cents.
Imho the integration means two major things:
 1) autotyping
 2) intercepting the "Login" action and offering to save/update the record in the base.
I (and I believe, many if not the most of users) do really need both of them.

**Comment by Sigmanor** [comment]
I think keepass-http will be enough.

**Comment by borisrorsvort** [comment]
:+1:

**Comment by vbrajon** [comment]
:+1:

**Comment by olebedev** [comment]
:+1:

**Comment by vla7** [comment]
:+1:

**Comment by Sispheor** [comment]
:+1:

**Comment by stuarta0** [comment]
:+1:

**Comment by marcopelegrini** [comment]
:+1:

**Comment by floviolleau** [comment]
:+1:

**Comment by PaulAnnekov** [comment]
@floviolleau @marcopelegrini stop doing this! Subscribed users will get your totally useless messages to their emails. Add emoji to topic starter's message instead.

**Comment by antelle** [comment]
Quick status update:
- auto-type will be implemented in v1.2 as a hotkey and global hotkey and will be already available in beta version soon
- integration with browser plugins is planned to v1.3

**Note**: @PaulAnnekov is absolutely right

**Comment by pedzed** [comment]
Any updates on this?

**Comment by antelle** [comment]
For now, auto-type is available with url detection, which can be used instead of plugin integration in some sense. For v1.4, this feature is not planned: we have to support new KeePass format for now.

**Comment by dasJ** [comment]
👍

**Comment by soredake** [comment]
https://github.com/luckyrat/KeeFox

**Comment by grinsted** [comment]
I noticed keepasshttp is no longer on the TODO list. Is that on purpose?

**Comment by Sigmanor** [comment]
sad if keepasshttp will not be added

**Comment by pwFoo** [comment]
Maybe use something like
* https://chrome.google.com/webstore/detail/ckp-keepass-integration-f/lnfepbjehgokldcaljagbmchhnaaogpc
* https://play.google.com/store/apps/details?id=com.hanhuy.android.keepshare.lite
as workaround?

**Comment by gynet** [comment]
Looking for having a browser extension to support auto-fill

**Comment by gynet** [comment]
Implementing a browser extension would be beneficial to many potential features for Keeweb, e.g auto fill credit card info and address,  auto saving password,

**Comment by robsonsobral** [comment]
I'm sorry for bother.

How can I use auto type? I tried the wiki, but no without luck. I'm sorry.

**Comment by antelle** [comment]
You can use auto-type in desktop apps using special key combinations, you can see your shortcuts in app settings, Shortcuts page.

**Comment by gynet** [comment]
I am currently using keepassxc as the workaround, and i think some people are using that as well. Once Keeweb enable this feature, many people will move over to KeeWeb

**Comment by antelle** [comment]
So, now we support plugins and it should be pretty easy to implement it.
Experimental KeeWebHttp plugin: https://plugins.keeweb.info/plugins/keewebhttp/
Please **DO NOT** use this version with your data! Uninstall the plugin after you try it. It's not ready, not secure and not stable. It's there as a POC.
But feel free to test on test db.

**Comment by gynet** [comment]
@antelle awesome!

**Comment by antelle** [comment]
Progress report:
- plugin is now secure, feel free to install it (works only with KeeWeb >= 1.5.1)
- it shares the configuration with KeePassHttp, settings are stored inside a special entry (called `KeePassHttp Settings`) and will be understood by both KeePass and KeeWeb

TODO, before we can publish it (I'll keep this post up to date):
- [ ] entry creation
- [ ] lock/unlock/focus: investigate usability and implement
- [ ] maybe some settings, but definitely we'll not have them like in keepasshttp

If anyone wants to test, please let me know. The plugin will be added to the plugin gallery next week. Now you can scroll to the bottom of Plugins page and install it from this url: https://plugins.keeweb.info/plugins/keewebhttp/

**Comment by antelle** [comment]
Finally issues mentioned above are resolved, the plugin is available in KeeWeb plugin gallery:
![plugins](https://cloud.githubusercontent.com/assets/633557/26462403/ebdc3de0-4180-11e7-8805-710fc574ae28.png)
Please let me know if you find any issues with it.

**Comment by snakyjake1** [comment]
I don't see the option for keewebhttp when using the web app:
I'm using https://app.keeweb.info/

I see the language plugins (and wish I could hide them).

**Comment by antelle** [comment]
Of course, it's desktop plugin.

**Comment by ericbn** [comment]
Thank you! Works beautifully with both chromeIPass and PassIFox on Mac! 🎉🎉🎉

**Comment by vla7** [comment]
Great! Thank you!

**Comment by navdeepio** [comment]
how do I set it up? there's no option in keeweb desktop for installing plugins..

**Comment by SavageCore** [comment]
@navdeepio On the left of the Settings page under Shortcuts. Just install it then go back to browser and it'll ask to connect.

**Comment by antelle** [comment]
Locking to prevent further discussions here, as all major issues are now resolved.
Please feel free to open a new issue, if you find and problem with keewebhttp.
